server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java

```diff
@@ -1622,4 +1627,170 @@ public void testRealtimeMultiGetRequestsUnsuccessful() {
+        assertThat(fields.size(), equalTo(1));
+
+    }
+
+    public void testRealtimeTermVectorRequestsUnSuccessful() throws IOException {
+        final String primary = internalCluster().startDataOnlyNode();
+        XContentBuilder mapping = jsonBuilder().startObject()
+            .startObject("properties")
+            .startObject("field")
+            .field("type", "text")
+            .field("term_vector", "with_positions_offsets_payloads")
+            .field("analyzer", "tv_test")
+            .endObject()
+            .endObject()
+            .endObject();
+        // refresh interval disabled to ensure refresh rate of index (when data is ready for search) doesn't affect realtime termvectors
+        assertAcked(
+            prepareCreate(INDEX_NAME).setMapping(mapping)
+                .addAlias(new Alias("alias"))
+                .setSettings(
+                    Settings.builder()
+                        .put(indexSettings())
+                        .put("index.analysis.analyzer.tv_test.tokenizer", "standard")
+                        .put("index.refresh_interval", -1)
+                        .putList("index.analysis.analyzer.tv_test.filter", "lowercase")
+                        .put(indexSettings())
+                        .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 2)
+                )
+        );
+        final String replica = internalCluster().startDataOnlyNode();
+        ensureGreen(INDEX_NAME);
+        final String id = routingKeyForShard(INDEX_NAME, 0);
+        final String routingOtherShard = routingKeyForShard(INDEX_NAME, 1);
+
+        // index doc 1
+        client().prepareIndex(INDEX_NAME)
+            .setId(Integer.toString(1))
+            .setSource(
+                jsonBuilder().startObject()
+                    .field("field", "the quick brown fox jumps over the lazy dog")
+                    // 0the3 4quick9 10brown15 16fox19 20jumps25 26over30
+                    // 31the34 35lazy39 40dog43
+                    .endObject()
+            )
+            .setRouting(id)
+            .execute()
+            .actionGet();
+
+        // non realtime termvectors 1
+        TermVectorsResponse response = client().prepareTermVectors(indexOrAlias(), Integer.toString(1)).setRealtime(false).get();
+        assertFalse(response.isExists());
+
+        // realtime termvectors (preference = _replica)
+        TermVectorsRequestBuilder resp = client(replica).prepareTermVectors(indexOrAlias(), Integer.toString(1))
+            .setPayloads(true)
+            .setOffsets(true)
+            .setPositions(true)
+            .setPreference(Preference.REPLICA.type())
+            .setSelectedFields();
+        response = resp.execute().actionGet();
+
+        assertFalse(response.isExists());
+        assertThat(response.getIndex(), equalTo(INDEX_NAME));
+
+        // realtime termvectors (with routing set)
+        resp = client(replica).prepareTermVectors(indexOrAlias(), Integer.toString(1))
```

Comment:
Just to confirm - so when routing is set, we don't want to reroute to primary for TermVectors?

The way shouldForcePrimaryRouting exists right now, we never examine the routing parameter when we decide to reroute to primary.